pandas/tests/indexing/interval/test_interval_new.py

```diff
@@ -107,22 +104,22 @@ def test_loc_with_slices(self):
         result = s[Interval(0, 1):Interval(2, 3)]
         tm.assert_series_equal(expected, result)
 
-        expected = s.iloc[4:]
+        expected = s.iloc[3:]
         result = s.loc[Interval(3, 4):]
         tm.assert_series_equal(expected, result)
         result = s[Interval(3, 4):]
         tm.assert_series_equal(expected, result)
 
-        with pytest.raises(KeyError):
+        with pytest.raises(NotImplementedError):
```

Comment:
It feels a bit strange to return a NotImplementedError here. I would at least add a error message about slicing not being implemented for non-exact matches

Reply:
Done. Since the `NotImplementedError` comes from `_searchsorted_monotonic` I had to make the message a bit more generic, as it could conceivably be raised in other scenarios.